packages/mui-system/src/createStyled/createStyled.js

```diff
@@ -98,14 +95,15 @@ export default function createStyled(input = {}) {
     slotShouldForwardProp = shouldForwardProp,
   } = input;
 
-  const systemSx = (props) => {
-    return styleFunctionSx(attachTheme(props, themeId, defaultTheme));
-  };
-  systemSx.__mui_systemSx = true;
+  function styleAttachTheme(props) {
+    attachTheme(props, themeId, defaultTheme)
+  }
 
   const styled = (tag, inputOptions = {}) => {
-    // Filter out the `sx` style function from the previous styled component to prevent unnecessary styles generated by the composite components.
-    processStyles(tag, (styles) => styles.filter((style) => !style?.__mui_systemSx));
+
+    // If `tag` is already a styled component, filter out the `sx` style function
+    // to prevent unnecessary styles generated by the composite components.
+    processStyles(tag, (styles) => styles.filter(style => style !== styleFunctionSx));
```

Comment:
Isn't this mutating `tag` though?

Reply:
@siriwatknp Maybe you could confirm here, I think this line is mutating `tag`'s style processors, but what we want is to not add the sx processor to this new styled component. Is that right?

Reply:
Regardless of the question above, I think I'll rename this function to `internal_mutateStyles` to better express what it's doing.

Reply:
I think this could be an (existing) issue/bug but I don't have the bandwidth to investigate it and I'm afraid of fixing it and possibly changing the behavior, so I've just renamed `internal_processStyles` to `internal_mutateStyles` to better highlight what is happening here.

Reply:
I think we introduced this as an perf improvement as we were processing sx multiple times if we had a styled component invoked over another styled component. This is what I remember about it. Likely @siriwatknp could have more info, but +1 on not changing any behavior as part of this PR.

Reply:
Yes, @mnajdova is right.